feat(ELB): add elb ipgroups data source (#3637)
huaweicloud/terraform-provider-huaweicloud: docs/data-sources/elb_ipgroups.md

---
subcategory: Dedicated Load Balance (Dedicated ELB)
---

# huaweicloud_elb_ipgroups

> Use this data source to get the list of ELB ipgroups.

Reviewer: please replace all **ipgroups** to **IP groups** in the docs, logs and error message.

## Example Usage

```hcl
variable "ipgroup_name" {}

data "huaweicloud_elb_ipgroups" "test" {
  name = var.ipgroup_name
}
```

## Argument Reference

The following arguments are supported:

* `region` - (Optional, String) Specifies the region in which to query the data source.
  If omitted, the provider-level region will be used.

* `name` - (Optional, String) Specifies the name of the IP address group.

* `ipgroup_id` - (Optional, String) Specifies the ID of the IP address group.

* `description` - (Optional, String) Specifies the description of the IP address group.

> * `ip_address` - (Optional, String) Specifies the IP address in the IP address group.

Reviewer 2: ```suggestion
* `ip_address` - (Optional, String) Specifies the IP address of the IP address group.
```

## Attribute Reference

In addition to all arguments above, the following attributes are exported:

* `id` - The data source ID.

* `ipgroups` - Lists the ipgroups.
  The [ipgroups](#Elb_ipgroups) structure is documented below.

<a name="Elb_ipgroups"></a>
The `ipgroups` block supports:

* `id` - The ID of the IP address group.

* `name` - The name of the IP address group.

> * `description` - Provides supplementary information about the IP address group.

Reviewer 2: ```suggestion
* `description` - The description of the IP address group.
```

* `project_id` - The project ID of the IP address group.

* `listeners` - The IDs of listeners with which the IP address group is associated. The [object](#Elb_ipgroups_listeners)
  structure is documented below.

* `ip_list` - The IP addresses or CIDR blocks in the IP address group. The [object](#Elb_ipgroups_ip_list) structure is

>   documented below.

Reviewer 2: ```suggestion
* `listeners` - The IDs of listeners with which the IP address group is associated. The [listeners](#Elb_ipgroups_listeners)
  structure is documented below.

* `ip_list` - The IP addresses or CIDR blocks in the IP address group. The [ip_list](#Elb_ipgroups_ip_list) structure is
  documented below.
```

* `created_at` - The time when the IP address group was created.

* `updated_at` - The time when the IP address group was updated.

<a name="Elb_ipgroups_listeners"></a>
The `listeners` block supports:

* `id` - The listener ID.

<a name="Elb_ipgroups_ip_list"></a>
The `ip_list` block supports:

> * `ip` - The IP addresses in the IP address group.
> 
> * `description` - The remarks about the IP address group.

Reviewer 2: ```suggestion
* `ip` - The IP address.

* `description` - The description of the IP address.
```

> * `description` - Provides remarks about the IP address group.

Reviewer 2: ```suggestion
* `ip` - The IP address.

* `description` - The description of the IP address.
```
